### PR TITLE
New operator failing on health checks

### LINKED ISSUE #30355: New operator failing on health checks
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

operator

### Describe the bug

I tried to migrate my image to the latest (25.0.0), then I installed the newest k8s operator.

But the pods would always be marked as unhealthy, and I could never reach the admin console.

The errors I got were these: 

```
 Startup probe failed: Get "http://10.244.197.15:9000/health/started": d │
│ ial tcp 10.244.197.15:9000: connect: connection refused                                                                              │
│   Warning  Unhealthy  80s (x12 over 91s)  kubelet            Startup probe failed: Get "http://10.244.197.15:9000/health/started": E │
│ OF
```

I don't know if this was enough to prevent access, but FWIW, I simply couldn't access the console or the clients.

After trying a few times, I tried to build an image with the hostname:v1 feature enabled. It didn't help.

I don't see a particular fail on the logs.

I had to rollback to an older operator and to 24.0.5.

### Version

25.0.0

### Regression

- [X] The issue is a regression

### Expected behavior

I expected to be able to access the console and to log in.

### Actual behavior

Pods don't report as healthy. Console inaccessible.

### How to Reproduce?

I installed the new operator, and built a newer image.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mabartos**
Closes #30355

### CODE REVIEW

**Comment by mabartos** [comment]
@keycloak/cloud-native I believe it's sufficient to resolve the parent issue. Are you okay with that, or do you have any suggestions?

File: docs/guides/operator/customizing-keycloak.adoc [hunk 1]
```diff
@@ -43,10 +43,11 @@ spec:
...
 
-[NOTE]
-====
-With custom images, every build time option passed either through a dedicated field or the `additionalOptions` is ignored.
-====
+NOTE: With custom images, every build time option passed either through a dedicated field or the `additionalOptions` is ignored.
+
+NOTE: Operator is unaware of any TLS settings in your `Dockerfile` when using a custom image.
+It may cause the management interface uses the `https` schema, but the Operator accesses it via `http`.
+To ensure proper TLS configuration, use the `tlsSecret` and `truststores` properties in the Keycloak CR.
```
Comment (vmuzikar) [inline docs/guides/operator/customizing-keycloak.adoc:50]: I'd maybe make it more generic. Operator is unaware of any config options baked in a custom image. This is problematic namely TLS and HTTP(S) ports settings which Operator uses when configuring Services.
WDYT?

**Comment by mabartos** [comment]
@vmuzikar Might be better now. PTAL

**Review by vmuzikar — COMMENTED** [review]
@mabartos Thank you for this docs enhancement!

Added some suggestions to the wording but I'm not a native speaker so I don't consider it blocking. :D

**Comment by mabartos** [comment]
@vmuzikar Thanks! Looks better :)) Updated

**Review by vmuzikar — APPROVED** [review]

**Pull request merged**